Tests/SwiftDocCTests/Semantics/Reference/TabNavigatorTests.swift

```diff
@@ -0,0 +1,202 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+import XCTest
+@testable import SwiftDocC
+import Markdown
+
+class TabNavigatorTests: XCTestCase {
+    func testNoTabs() throws {
+        let (renderBlockContent, problems, tabNavigator) = try parseDirective(TabNavigator.self) {
+            """
+            @TabNavigator
+            """
+        }
+        
+        XCTAssertNotNil(tabNavigator)
+        
+        XCTAssertEqual(
+            problems,
+            ["1: warning – org.swift.docc.HasAtLeastOne<TabNavigator, Tab>"]
+        )
+        
+        XCTAssertEqual(renderBlockContent.count, 1)
+        XCTAssertEqual(
+            renderBlockContent.first,
+            .tabNavigator(.init(tabs: []))
+        )
+    }
+    
+    func testEmptyTab() throws {
+        let (renderBlockContent, problems, tabNavigator) = try parseDirective(TabNavigator.self) {
+            """
+            @TabNavigator {
+                @Tab("hiya") {
+                    
+                }
+            }
+            """
+        }
+        
+        XCTAssertNotNil(tabNavigator)
+        XCTAssertEqual(
+            problems,
+            ["2: warning – org.swift.docc.Tab.HasContent"]
+        )
+        
+        XCTAssertEqual(renderBlockContent.count, 1)
+        XCTAssertEqual(
+            renderBlockContent.first,
+            .tabNavigator(RenderBlockContent.TabNavigator(
+                tabs: [RenderBlockContent.TabNavigator.Tab(title: "hiya", content: [])]
+            ))
+        )
+        
+    }
+    
+    func testInvalidParametersAndContent() throws {
+        let (renderBlockContent, problems, tabNavigator) = try parseDirective(TabNavigator.self) {
+            """
+            @TabNavigator(tabs: 3) {
+                @Tab("hi") {
+                    Hello there.
+                }
+            
+                @Tab("hey", bestTab: true) {
```

Comment:
Not sure this kind of test belongs in here. It's more checking the directive parser than the tab navigator directive itself. I think as you already went through so much effort to make directive arguments to content so declarative IMHO there is no point checking random keys like that.

Reply:
Hm yeah that's fair. I'm asserting on the generated RenderBlockContent when an unknown directive is encountered as well below which is a little more interesting but I agree that `bestTab: true` is kind of useless at this point. I can remove it.

Reply:
The unknown directive case is more interesting in the sense that it maps more closely to the structure we expect from the content specifically for tab navigators.